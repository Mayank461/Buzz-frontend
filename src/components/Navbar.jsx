import axios from 'axios';
import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { API_URL } from '../config';

export default function Navbar() {
  const [user, setUser] = useState({});
  useEffect(() => {
    axios
      .get(`${API_URL}/user/getUser`, { withCredentials: true })
      .then((res) => {
        setUser(res.data[0]);
        console.log(user);
      })
      .catch((err) => console.log(err.message));
  }, []);

  function handleLogout() {
    window.open(`${API_URL}/auth/logout`, '_self');
  }
  return (
    <>
      <nav className="navbar navbar-light bg-light p-0">
        <div className="container-fluid">
          <div>
            <Link to={'/'}>
              <img
                className="logo"
                src="https://mma.prnewswire.com/media/728150/TO_THE_NEW_Logo.jpg?p=facebook"
              />
            </Link>
          </div>
          <div className="d-flex position-relative">
            <Link to='/profile' className='nav-link text-dark'>
            <div className='d-flex'>
            <div className="">
              <img
                src={user.picture_url}
                className="card-img-top small-round-pic  round-img"
                alt="..."
              />
            </div>
            <div className="d-flex align-items-center ms-2">
              {user.firstname + ' ' + user.lastname}
            </div>
            </div>
            </Link>
          
            <div className="d-flex align-items-center ms-2 round-img border rounded-circle icon-bg text-dark p-2">
              <i className="fa-brands fa-facebook-messenger "></i>
            </div>
            <div className="position-relative d-flex">
              <div className="d-flex align-items-center ms-2 round-img border rounded-circle icon-bg text-dark p-2 ">
             
                  <i className="fa-solid fa-user"></i>
             
              </div>
              <div className="round-img bg-danger p-1 text-white incoming position-absolute bottom-50 end-0">
                1
              </div>
            </div>
            <div
              onClick={handleLogout}
              className="d-flex align-items-center ms-2 round-img border rounded-circle icon-bg text-dark p-2 pointer"
            >
              <i className="fa fa-sign-out "></i>
            </div>
          </div>
        </div>
      </nav>
    </>
  );
}
