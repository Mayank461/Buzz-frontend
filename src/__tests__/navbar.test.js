import { render, screen } from '@testing-library/react';
import Navbar from '../components/Navbar';
import { MemoryRouter as Router } from 'react-router-dom';

describe('Navbar test', () => {
  it('Navbar render with user data', () => {
    const user = {
      picture_url: 'https://user.com/img.png',
      firstname: 'John',
      lastname: 'Doe',
      friends: { myFriendRequests: [0, 0, 0] },
    };

    render(
      <Router>
        <Navbar user={user} />
      </Router>
    );

    expect(screen.queryByTitle('displayname')).toHaveTextContent(
      user.firstname + ' ' + user.lastname
    );
    expect(screen.getByTitle('friend_req_count')).toHaveTextContent(
      user.friends.myFriendRequests.length
    );
    expect(screen.getByTitle('logout-btn')).toBeTruthy();
  });

  it('Navbar render without user data', () => {
    const user = {
      friends: { myFriendRequests: [] },
    };

    render(
      <Router>
        <Navbar user={user} />
      </Router>
    );

    expect(screen.queryByTitle('displayname')).toHaveTextContent(
      'Edit Profile'
    );
    expect(screen.getByTitle('friend_req_count')).toHaveTextContent(
      user.friends.myFriendRequests.length
    );
    expect(screen.getByTitle('logout-btn')).toBeTruthy();
  });
});
