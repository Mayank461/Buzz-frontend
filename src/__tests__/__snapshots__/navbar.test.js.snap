// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar test Navbar render check 1`] = `
<nav
  className="navbar  navbar-light bg-light p-0"
>
  <div
    className="container-fluid"
  >
    <div>
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="logo"
          className="logo"
          src="https://mma.prnewswire.com/media/728150/TO_THE_NEW_Logo.jpg?p=facebook"
        />
      </a>
    </div>
    <div
      className="d-flex position-relative"
    >
      <a
        className="nav-link text-dark"
        href="/profile"
        onClick={[Function]}
      >
        <div
          className="d-flex"
        >
          <div
            className=""
          >
            <img
              alt="..."
              className="card-img-top small-round-pic  round-img"
              src="https://user.com/img.png"
            />
          </div>
          <div
            className="d-flex align-items-center ms-2"
            title="displayname"
          >
            John Doe
          </div>
        </div>
      </a>
      <div
        className="d-flex align-items-center ms-2 round-img border rounded-circle icon-bg text-dark p-2"
      >
        <a
          className="text-dark"
          href="/"
          onClick={[Function]}
        >
          <i
            className="fa-solid fa-house"
          />
        </a>
      </div>
      <div
        className="position-relative d-flex "
      >
        <a
          className="d-flex align-items-center  ms-2 round-img border rounded-circle icon-bg text-dark p-2 "
          href="/friends"
          onClick={[Function]}
        >
          <i
            className="fa-solid fa-user"
          />
        </a>
        <div
          className="round-img bg-danger p-1 text-white incoming position-absolute bottom-50 end-0"
          title="friend_req_count"
        >
          3
        </div>
      </div>
      <div
        className="d-flex align-items-center ms-2 me-3 round-img border rounded-circle icon-bg text-dark p-2 pointer"
        onClick={[Function]}
        title="logout-btn"
      >
        <i
          className="fa fa-sign-out "
        />
      </div>
    </div>
  </div>
</nav>
`;
